receiver/nginxreceiver/scraper.go

```diff
@@ -16,13 +16,16 @@ package nginxreceiver
 
 import (
 	"context"
+	"fmt"
 	"net/http"
+	"strings"
 	"time"
 
 	"github.com/nginxinc/nginx-prometheus-exporter/client"
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/model/pdata"
 	"go.uber.org/zap"
+	"google.golang.org/grpc/codes"
```

Comment:
Why using gRPC codes?

Reply:
I think the idea here is to use a consistent non-string error code to denote common failures and diagnostics across receivers.

I can write up a proposal around this if you'd like.

Reply:
Please do, but in Golang this is usually what "error" does. We can define custom errors as we already did for "scrapererror" :) No need to use "grpc" codes here I think.

Reply:
Sorry for long delay here.   Yes, I think doing what was done for scrapererror is exactly what would help.   Just proposed gRPC codes for expediency since they're used in OTLP.

Reply:
Looking at details, I think we're missing something in scrapererror, wrote more down here: https://github.com/open-telemetry/opentelemetry-collector/issues/4156